File: src/main/java/com/herringbone/stock/model/MonthlyBucket.java
package com.herringbone.stock.model;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.herringbone.stock.util.CustomDateTimeSerializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.Transient;
import java.time.ZonedDateTime;

@Entity
@Table(name="monthlybuckets")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MonthlyBucket {
    @EmbeddedId
    @JsonSerialize
    MonthlyBucketId monthlyBucketId;

    @Column(name="cumulativepercentage")
    private double cumulativePercentage;

    @Column(name="cumulativefrequency")
    private double cumulativefrequency;

    @Column(name="frequency")
    private double frequency;

    @Column(name="percentage")
    private double percentage;

    @Transient
    @JsonSerialize(using = CustomDateTimeSerializer.class)
    private ZonedDateTime lastStoredQuoteDate;

    @Transient
    @JsonSerialize
    private Double lastStoredQuoteClose;
}

File: src/main/js/bucket/MonthlyBucket.js
import React from "react";
var convertTrendType = require('../util.js').convertTrendType;

class MonthlyBucket extends React.Component {
    render() {
        return (
            <tr>
                <td>Monthly</td>
                <td>{convertTrendType(this.props.monthlyBucket.monthlyBucketId.trendType)}</td>
                <td>{this.props.monthlyBucket.monthlyBucketId.monthsInTrendCount}</td>
                <td>{this.props.monthlyBucket.cumulativePercentage}</td>
                <td>{this.props.monthlyBucket.lastStoredQuoteClose}</td>
                <td>{this.props.monthlyBucket.lastStoredQuoteDate}</td>
            </tr>
        )
    }
}

export default MonthlyBucket;
